HymnsTests/Display/DisplayHymnViewModelSpec.swift

```diff
@@ -100,38 +89,25 @@ class DisplayHymnViewModelSpec: QuickSpec {
                                 }
-                                it("should have lyrics") {
-                                    expect(target.lyrics).to(equal([Verse]()))
-                                }
                                 let chordsUrl = URL(string: "http://www.hymnal.net/en/hymn/c/1151/f=gtpdf")!
-                                it("chords url should be \(String(describing: chordsUrl))") {
-                                    expect(target.chordsUrl).to(equal(chordsUrl))
-                                }
                                 it("chords url should be prefetched") {
-                                    verify(webViewPreloader.preload(url: chordsUrl)).wasCalled(exactly(1))
+                                    verify(pdfLoader.load(url: chordsUrl)).wasCalled(exactly(1))
                                 }
                                 let guitarUrl = URL(string: "http://www.hymnal.net/en/hymn/c/1151/f=pdf")!
-                                it("guitar url should be \(String(describing: guitarUrl))") {
-                                    expect(target.guitarUrl).to(equal(guitarUrl))
-                                }
                                 it("guitar url should be prefetched") {
-                                    verify(webViewPreloader.preload(url: guitarUrl)).wasCalled(exactly(1))
+                                    verify(pdfLoader.load(url: guitarUrl)).wasCalled(exactly(1))
                                 }
                                 let pianoUrl = URL(string: "http://www.hymnal.net/en/hymn/c/1151/f=ppdf")!
-                                it("piano url should be \(String(describing: pianoUrl))") {
-                                    expect(target.pianoUrl).to(equal(pianoUrl))
-                                }
                                 it("piano url should be prefetched") {
-                                    verify(webViewPreloader.preload(url: pianoUrl)).wasCalled(exactly(1))
+                                    verify(pdfLoader.load(url: pianoUrl)).wasCalled(exactly(1))
                                 }
                             }
                         }
                     }
                     context("for new song 145") {
                         beforeEach {
-                            target = DisplayHymnViewModel(backgroundQueue: testQueue, hymnToDisplay: newSong145, hymnsRepository: hymnsRepository,
-                                                          mainQueue: testQueue, favoritesStore: favoritesStore, historyStore: historyStore,
-                                                          webviewCache: webViewPreloader)
+                            target = DisplayHymnViewModel(analytics: analytics, backgroundQueue: testQueue, favoritesStore: favoritesStore, hymnToDisplay: newSong145, hymnsRepository: hymnsRepository,
```

Comment:
Here you're missing the HistoryStore.

The raywenderlich article you sent me shows how to run these tests one at a time. https://www.raywenderlich.com/135-behavior-driven-testing-tutorial-for-ios-with-quick-nimble#toc-anchor-007
You can usually use that to isolate which tests are throwing the errors so you can figure out which tests to focus on.

Reply:
I guess I deleted History store and the HymnsRepository without realizing it. Thanks...